test/spec/test-stall-detection.js

```diff
@@ -0,0 +1,425 @@
+import { Upload } from 'tus-js-client'
+import { TestHttpStack, getBlob, waitableFunction } from './helpers/utils.js'
+
+// Custom HTTP stack that can simulate stalls
+class StallSimulatingHttpStack extends TestHttpStack {
+  constructor() {
+    super()
+    this.stallOnNextPatch = false
+  }
+
+  createRequest(method, url) {
+    const req = super.createRequest(method, url)
+
+    // If we want to simulate a stall on PATCH requests, override the send method
+    if (this.stallOnNextPatch && method === 'PATCH') {
+      this.stallOnNextPatch = false
+      req.send = async function (body) {
+        this.body = body
+        if (body) {
+          this.bodySize = await getBodySize(body)
+          // Don't call progress handler to simulate a stall
+        }
+        this._onRequestSend(this)
+        return this._requestPromise
+      }
+    }
+
+    return req
+  }
+
+  supportsProgressEvents() {
+    return true
+  }
+}
+
+// Helper to get body size
+function getBodySize(body) {
+  if (body == null) return null
+  if (body instanceof Blob) return body.size
+  if (body.length != null) return body.length
+  return 0
+}
+
+describe('tus-stall-detection', () => {
+  describe('configuration options', () => {
```

Comment:
I greatly appreciate you writing these configuration tests, but I am wondering if they are truly necessary here. They mostly seem to check that the option merging is working properly. Bug in there would also be detected in the integration tests, so I think we can use some LOC here :)